### PR TITLE
Modal generalization

### PR DESCRIPTION
**Pull request by nkenneweg**
Modal Generalizations to enable other plugins to use the modal.
Also cherry-picked @GarstgerUnhold bugfix.
This might still need work. If you want to merge this, proceed with caution!

### CODE REVIEW

**Comment by tessi** [comment]
Tests are failing :'(

**Comment by mfrister** [comment]
Is there an openproject.org ticket for this one? Doesn't seem to be a small change, so I'd like one.
Once you have an OP ticket number, please add a line to the changelog.

**Comment by tessi** [comment]
Is this still WIP? Can't find an issue on openproject.org for this PR.

**Comment by apfelfabrik** [comment]
How about this one? https://www.openproject.org/issues/1429

**Comment by tessi** [comment]
yay, this is what I'm looking for

**Comment by apfelfabrik** [comment]
Watching in modals is currently broken for planning elements. Can only reproduce in one environment, not in another, will investigate.

Furthermore, I am not sure about the client-side removal of decoration DOM elements, i.e., the top and left menu. Doing this on the client-side currently leads to flickering of the whole page rendered with layout and then the without. Perhaps there is a way to fix this without changing all the views.

How about putting the menu removal and link alteration code into the server-rendered page and have it execute at load time? We could also render everything invisible and make it visible once all alterations are performed. We could also move everything to the server, which should result in the best performance.

I would really appreciate more feedback on this. Guys?

**Comment by apfelfabrik** [comment]
Ok, watchers were my fault. Works as advertised!

**Comment by apfelfabrik** [comment]
Layout should probably have a different name. How about decoration=false? Opinions? Not a show-stopper right now.

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -488,21 +488,29 @@ document.observe("dom:loaded", function() {
...
 
+      request.options.requestHeaders['X-Authentication-Scheme'] = "Session";
+
       if ($('ajax-indicator') && Ajax.activeRequestCount > 0) {
         Element.show('ajax-indicator');
       }
     },
-    onComplete: function(){
+    onComplete: function(request, result){
+      if (result.status === 401 && /Reason: login needed/.match(result.getAllResponseHeaders())) {
```
Comment (apfelfabrik) [inline app/assets/javascripts/application.js]: Still not a fan of "Reason"-Header. At least prefix an X.

File: app/assets/javascripts/timelines_modal.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/** create a planning modal
```
Comment (apfelfabrik) [inline app/assets/javascripts/timelines_modal.js]: New files need the copyright header.

File: app/views/layouts/base.html.erb [hunk 3]
```diff
@@ -98,9 +101,11 @@ See doc/COPYRIGHT.rdoc for more details.
     </div>
   </div>
 
-  <div id="footer">
-    <%= footer_content %>
-  </div>
+  <% if (show_menu) %>
```
Comment (apfelfabrik) [inline app/views/layouts/base.html.erb]: if show_menu show footer?

**Pull request merged**